util/PaginationEmbed.js

```diff
@@ -0,0 +1,564 @@
+const { MessageEmbed } = require("discord.js");
+
+/**
+ * @extends {MessageEmbed}
+ */
+class PaginationEmbed extends MessageEmbed {
+  /**
+   * Options for PaginationEmbed.clientMessage.
+   * @typedef {Object} ClientMessageOptions
+   * @prop {Object} [message=null] - The message object sent by the client, if there is any.
+   * @prop {String} [content="Preparing..."] - The custom message content while preparing the embed.
```

Comment:
The types `String`, `Boolean`, `Number` refer to their constructors rather than their primitives. Please change them to `string`, `boolean` and `number`, respectively.